src/modules/mavlink/mavlink_messages.cpp

```diff
@@ -597,7 +598,7 @@ class MavlinkStreamSysStatus : public MavlinkStream
 
 			for (unsigned int i = 0; i < (sizeof(bat_msg.voltages) / sizeof(bat_msg.voltages[0])); i++) {
 				if ((int)i < battery_status.cell_count && battery_status.connected) {
-					bat_msg.voltages[i] = (battery_status.voltage_v / battery_status.cell_count) * 1000.0f;
+					bat_msg.voltages[i] = battery_status.voltage_cell_v[i] * 1000.0f;
```

Comment:
This assumes that cell voltages are valid.

Reply:
The old way just takes the pack voltage and divides by 4.

Reply:
That's not the point. For a non-SMBus equipped drone you're now potentially feeding garbage to the GCS. Have you verified that the individual voltages are all valid for the ADC-driven battery readout?